src/operator/tensor/indexing_op.h

```diff
@@ -598,7 +598,12 @@ void EmbeddingOpBackward(const nnvm::NodeAttrs& attrs,
         uint64_t shape_out_prod =
           static_cast<uint64_t>(grad_out.shape_[0])*
           static_cast<uint64_t>(grad_out.shape_[1]);
-        if (shape_out_prod < (uint64_t)16384 && shape_in_prod < (uint64_t)16384) {
+
+        const char *type = getenv("MXNET_FORCE_ADDTAKEGRAD");
+        const bool default_addtakegrad = (type == nullptr);
```

Comment:
shall we do this in constructor and keep it as a private variable instead?

Reply:
Operator doesn't have a class. Suggest to `dmlc::GetEnv` and make the result static so lookup is done only once.